service_crategen/src/commands/generate/codegen/mod.rs

```diff
@@ -165,7 +165,7 @@ where
     // See https://github.com/rusoto/rusoto/issues/519
     writeln!(writer,
              "/// Trait representing the capabilities of the {service_name} API. {service_name} clients implement this trait.
-        pub trait {trait_name} {{
+        pub trait {trait_name}: region::GetRegion {{
```

Comment:
I think this is great. I wonder if this wouldn't be a good opportunity to make this kind of change simpler in the future.

I could see a case made for this traits to declare a relation to a more general `rusoto_core::Service` trait which for now would be an empty trait thought could itself declare a relation to `region::GetRegion`

in rusoto_core

`pub trait Service: region::GetRegion { }`

in service crate

` pub trait {trait_name}: rusoto_core::Service`

What this would allow for is only needing to do this kind of trait extension once and in the future you make changes in more or less one place.

Thoughts?

Reply:
👍 that looks like a good way to allow future extension. How about trying that in a future PR, perhaps when we've a concrete need/ask for it.

Reply:
@softprops Thinking a bit about this - I'd like to suggest lumping together `Service` and `GetRegion` traits, rather than building a dependency chain. Then `Service` trait will be a base for all the other services and will provide the common functionality. `self.region()` being just one of them. How does that sound?